magicbyirineu/Persistence/Entities/CardDao.swift

```diff
@@ -0,0 +1,33 @@
+import Realm
+import RealmSwift
+
+class CardDao: Object {
+    @objc dynamic var name: String = ""
+    @objc dynamic var set: String = ""
+    @objc dynamic var setName: String = ""
+    @objc dynamic var imageUrl: String?
+    let types = List<RealmString>()
+    let foreignNames = List<ForeignNamesDao>()
+
+    @objc dynamic var uniqueIdentifierId = UUID().uuidString
+
+    override static func primaryKey() -> String? {
+        return "uniqueIdentifierId"
```

Comment:
```suggestion
        return uniqueIdentifierId
```

você n quis dizer que ele seria igual a isso aqui?

` @objc dynamic var uniqueIdentifierId = UUID().uuidString`

Reply:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/47391973/54361361-9f407480-4645-11e9-9f59-9ab51ccfb8d9.png">
Esta é a forma como a primary key é implementada no realm, saca só a imagem acima da documentação deles, é uma forma de dizer qual coluna da tabela é selecionada ao buscar a chave primária daquele registro

Reply:
Se este método não for implementado eu poderia cadastrar duas cartas iguais

Reply:
aaaaaah ta... Entendido. Show @cs-daniel-lima .